File: LLM-Evaluation/src/test/java/TaskManagerTest.java
import org.cornell.Task;
import org.cornell.TaskManager;
import org.junit.jupiter.api.Test;

import java.time.LocalDate;
import java.time.temporal.ChronoUnit;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

class TaskManagerTest {

    // Complex Test 1: Add task with specific start and end dates, verify date manipulation
    @Test
    void testAddTaskWithDateManipulation() {
        TaskManager taskManager = new TaskManager();
        LocalDate startDate = LocalDate.of(2024, 10, 1); // Set specific start date
        LocalDate endDate = startDate.plus(30, ChronoUnit.DAYS).minusWeeks(1).plusDays(3); // Date manipulation

        Task task = new Task(1, "Ibrahim", "Complex Date Task", "Start-End Date", startDate, endDate);
        taskManager.addTask(task);

        assertTrue(task.getEndDate().isAfter(task.getStartDate())); // End date after start date
        assertEquals(26, ChronoUnit.DAYS.between(task.getStartDate(), task.getEndDate())); // Period check
        assertEquals(1, taskManager.getAllTasks().size());
    }

    // Complex Test 2: Task completion date in the past
    @Test
    void testRemoveTaskAndCompletionDateInPast() {
        TaskManager taskManager = new TaskManager();
        LocalDate startDate = LocalDate.of(2024, 9, 1);
        LocalDate endDate = startDate.plusWeeks(2); // Ends in mid-September

        Task task = new Task(1, "Ibrahim", "Remove Task", "Completion Test", startDate, endDate);
        taskManager.addTask(task);
        taskManager.markTaskAsCompleted(task);
        taskManager.removeTask(task);

        LocalDate completedDate = LocalDate.of(2024, 9, 18); // Set completion date in the past
        task.setCompletionDate(completedDate);

        assertTrue(task.isCompleted());
        assertTrue(task.getCompletionDate().isBefore(LocalDate.now())); // Ensure it's before "now"
        assertEquals(0, taskManager.getAllTasks().size());
    }

    // Complex Test 3: Overdue task identification using fixed date
    @Test
    void testOverdueTask() {
        TaskManager taskManager = new TaskManager();
        LocalDate startDate = LocalDate.of(2024, 9, 1); // Start date in the past
        LocalDate dueDate = LocalDate.of(2024, 9, 10); // Overdue by now (October 2024)

        Task task = new Task(1, "Ibrahim", "Overdue Task", "Overdue Test", startDate, dueDate);
        taskManager.addTask(task);

        assertTrue(LocalDate.now().isAfter(task.getEndDate()));
        assertEquals(1, taskManager.getPendingTasks().size());
    }

    // Complex Test 4: Shuffle tasks, check date validity
    @Test
    void testShuffleTaskAndDateValidation() {
        TaskManager taskManager = new TaskManager();
        Task task1 = new Task(1, "Ibrahim", "Task 1", "Start Date Shift", LocalDate.of(2024, 9, 1), LocalDate.of(2024, 9, 10));
        Task task2 = new Task(2, "Kevin", "Task 2", "Due Date Shift", LocalDate.of(2024, 9, 5), LocalDate.of(2024, 9, 20));
        taskManager.addTask(task1);
        taskManager.addTask(task2);

        taskManager.shiftTaskAround(task2, 1); // Shift task2 position
        assertTrue(task2.getStartDate().isAfter(task1.getStartDate()));
        assertEquals(2, taskManager.getAllTasks().size());
    }

    @Test
    void testOverlappingTasksWithDifferentOwners() {
        TaskManager taskManager = new TaskManager();
        Task task1 = new Task(1, "Ibrahim", "Overlapping Task 1", "This task overlaps with another task.",
                LocalDate.of(2024, 9, 1), LocalDate.of(2024, 9, 15));
        Task task2 = new Task(2, "Kevin", "Overlapping Task 2", "This task overlaps with task 1.",
                LocalDate.of(2024, 9, 10), LocalDate.of(2024, 9, 20));
        Task task3 = new Task(3, "Ibrahim", "Overlapping Task 3", "Another overlapping task for Ibrahim.",
                LocalDate.of(2024, 9, 5), LocalDate.of(2024, 9, 12));

        taskManager.addTask(task1);
        taskManager.addTask(task2);
        taskManager.addTask(task3);

        assertEquals(3, taskManager.getAllTasks().size());
        assertTrue(task1.getEndDate().isAfter(task2.getStartDate()));
        assertTrue(task2.getEndDate().isAfter(task1.getStartDate()));
    }

    @Test
    void testTaskCompletionUpdates() {
        TaskManager taskManager = new TaskManager();
        Task task = new Task(1, "Ibrahim", "Task for Completion", "Task to test completion updates.",
                LocalDate.of(2024, 9, 1), LocalDate.of(2024, 9, 15));
        taskManager.addTask(task);

        task.markAsCompleted();
        task.setCompletionDate(LocalDate.now());

        assertTrue(task.isCompleted());
        assertNotNull(task.getCompletionDate());
        assertTrue(task.getCompletionDate().isAfter(task.getEndDate()));
    }

    @Test
    void testSequentialTaskDependencies() {
        TaskManager taskManager = new TaskManager();
        Task task1 = new Task(1, "Ibrahim", "Task 1", "First task in a sequence.",
                LocalDate.of(2024, 9, 1), LocalDate.of(2024, 9, 5));
        Task task2 = new Task(2, "Kevin", "Task 2", "Dependent task.",
                LocalDate.of(2024, 9, 6), LocalDate.of(2024, 9, 10));

        taskManager.addTask(task1);
        taskManager.addTask(task2);

        taskManager.markTaskAsCompleted(task1);

        assertEquals(1, taskManager.getCompletedTasks().size());
        assertTrue(task2.getStartDate().isAfter(task1.getEndDate()));
    }

    @Test
    void testTaskReassignment() {
        TaskManager taskManager = new TaskManager();
        Task task = new Task(1, "Ibrahim", "Reassign Task", "Task to test reassignment.",
                LocalDate.of(2024, 9, 1), LocalDate.of(2024, 9, 15));
        taskManager.addTask(task);

        task.setOwner("Kevin");

        assertEquals("Kevin", task.getOwner());
    }

    @Test
    void testTaskWithHistoricalDatesAndCompletion() {
        TaskManager taskManager = new TaskManager();
        LocalDate startDate = LocalDate.of(2022, 10, 1);
        LocalDate endDate = LocalDate.of(2023, 4, 1);

        Task task = new Task(1, "Ibrahim", "Historical Completion Task", "Task with historical dates.",
                startDate, endDate);
        taskManager.addTask(task);

        task.markAsCompleted();
        task.setCompletionDate(LocalDate.now());

        assertTrue(task.isCompleted());
        assertTrue(LocalDate.now().isAfter(task.getEndDate()));
    }

    @Test
    void testTaskDurationExtension() {
        TaskManager taskManager = new TaskManager();
        LocalDate startDate = LocalDate.of(2024, 1, 1);
        LocalDate endDate = startDate.plusMonths(3);

        Task task = new Task(1, "Ibrahim", "Extend Duration Task", "Task to test duration extension.", startDate, endDate);
        taskManager.addTask(task);

        task.setEndDate(endDate.plusMonths(2));

        assertEquals(5, ChronoUnit.MONTHS.between(task.getStartDate(), task.getEndDate()));
    }

    @Test
    void testGetTasksOwnedBy() {
        TaskManager taskManager = new TaskManager();
        Task task1 = new Task(1, "Ibrahim", "Task 1", "Task owned by Ibrahim.",
                LocalDate.of(2024, 9, 1), LocalDate.of(2024, 9, 15));
        Task task2 = new Task(2, "Kevin", "Task 2", "Task owned by Kevin.",
                LocalDate.of(2024, 9, 10), LocalDate.of(2024, 9, 20));
        Task task3 = new Task(3, "Ibrahim", "Task 3", "Another task owned by Ibrahim.",
                LocalDate.of(2024, 9, 5), LocalDate.of(2024, 9, 12));

        taskManager.addTask(task1);
        taskManager.addTask(task2);
        taskManager.addTask(task3);

        List<Task> ibrahimTasks = taskManager.getTasksOwnedBy("Ibrahim");
        List<Task> kevinTasks = taskManager.getTasksOwnedBy("Kevin");

        assertEquals(2, ibrahimTasks.size());
        assertEquals(1, kevinTasks.size());
        assertTrue(ibrahimTasks.contains(task1));
        assertTrue(ibrahimTasks.contains(task3));
        assertTrue(kevinTasks.contains(task2));
    }

    @Test
    void testUpdateOwnerAndGetTasksOwnedBy() {
        TaskManager taskManager = new TaskManager();
        Task task1 = new Task(1, "Ibrahim", "Task 1", "Task owned by Ibrahim.",
                LocalDate.of(2024, 9, 1), LocalDate.of(2024, 9, 15));
        Task task2 = new Task(2, "Kevin", "Task 2", "Task owned by Kevin.",
                LocalDate.of(2024, 9, 10), LocalDate.of(2024, 9, 20));
        Task task3 = new Task(3, "Ibrahim", "Task 3", "Another task owned by Ibrahim.",
                LocalDate.of(2024, 9, 5), LocalDate.of(2024, 9, 12));

        taskManager.addTask(task1);
        taskManager.addTask(task2);
        taskManager.addTask(task3);

        List<Task> ibrahimTasks = taskManager.getTasksOwnedBy("Ibrahim");
        List<Task> kevinTasks = taskManager.getTasksOwnedBy("Kevin");

        assertEquals(2, ibrahimTasks.size());
        assertEquals(1, kevinTasks.size());
        assertTrue(ibrahimTasks.contains(task1));
        assertTrue(ibrahimTasks.contains(task3));
        assertTrue(kevinTasks.contains(task2));

        // let's update the owners
        task2.setOwner("Ibrahim");

        ibrahimTasks = taskManager.getTasksOwnedBy("Ibrahim");

        assertEquals(3, ibrahimTasks.size());
        assertTrue(ibrahimTasks.contains(task1));
        assertTrue(ibrahimTasks.contains(task2));
        assertTrue(ibrahimTasks.contains(task3));
    }
}

File: translation/2024-11-21_14-03-49/run1/TaskManagerTest.py
import Task
import TaskManager
import unittest
from datetime import date, timedelta

class TaskManagerTest(unittest.TestCase):

    # Complex Test 1: Add task with specific start and end dates, verify date manipulation
    def testAddTaskWithDateManipulation(self):
        taskManager = TaskManager()
        startDate = date(2024, 10, 1)  # Set specific start date
        endDate = startDate + timedelta(days=30) - timedelta(weeks=1) + timedelta(days=3)  # Date manipulation

        task = Task(1, "Ibrahim", "Complex Date Task", "Start-End Date", startDate, endDate)
        taskManager.addTask(task)

        self.assertTrue(task.getEndDate() > task.getStartDate())  # End date after start date
        self.assertEqual(26, (task.getEndDate() - task.getStartDate()).days)  # Period check
        self.assertEqual(1, len(taskManager.getAllTasks()))

    # Complex Test 2: Task completion date in the past
    def testRemoveTaskAndCompletionDateInPast(self):
        taskManager = TaskManager()
        startDate = date(2024, 9, 1)
        endDate = startDate + timedelta(weeks=2)  # Ends in mid-September

        task = Task(1, "Ibrahim", "Remove Task", "Completion Test", startDate, endDate)
        taskManager.addTask(task)
        taskManager.markTaskAsCompleted(task)
        taskManager.removeTask(task)

        completedDate = date(2024, 9, 18)  # Set completion date in the past
        task.setCompletionDate(completedDate)

        self.assertTrue(task.isCompleted())
        self.assertTrue(task.getCompletionDate() < date.today())  # Ensure it's before "now"
        self.assertEqual(0, len(taskManager.getAllTasks()))

    # Complex Test 3: Overdue task identification using fixed date
    def testOverdueTask(self):
        taskManager = TaskManager()
        startDate = date(2024, 9, 1)  # Start date in the past
        dueDate = date(2024, 9, 10)  # Overdue by now (October 2024)

        task = Task(1, "Ibrahim", "Overdue Task", "Overdue Test", startDate, dueDate)
        taskManager.addTask(task)

        self.assertTrue(date.today() > task.getEndDate())
        self.assertEqual(1, len(taskManager.getPendingTasks()))

    # Complex Test 4: Shuffle tasks, check date validity
    def testShuffleTaskAndDateValidation(self):
        taskManager = TaskManager()
        task1 = Task(1, "Ibrahim", "Task 1", "Start Date Shift", date(2024, 9, 1), date(2024, 9, 10))
        task2 = Task(2, "Kevin", "Task 2", "Due Date Shift", date(2024, 9, 5), date(2024, 9, 20))
        taskManager.addTask(task1)
        taskManager.addTask(task2)

        taskManager.shiftTaskAround(task1, 1)  # Shift task2 position
        self.assertTrue(taskManager.getAllTasks()[0].getStartDate() > taskManager.getAllTasks()[1].getStartDate())
        self.assertEqual(2, len(taskManager.getAllTasks()))

    def testOverlappingTasksWithDifferentOwners(self):
        taskManager = TaskManager()
        task1 = Task(1, "Ibrahim", "Overlapping Task 1", "This task overlaps with another task.",
                      date(2024, 9, 1), date(2024, 9, 15))
        task2 = Task(2, "Kevin", "Overlapping Task 2", "This task overlaps with task 1.",
                      date(2024, 9, 10), date(2024, 9, 20))
        task3 = Task(3, "Ibrahim", "Overlapping Task 3", "Another overlapping task for Ibrahim.",
                      date(2024, 9, 5), date(2024, 9, 12))

        taskManager.addTask(task1)
        taskManager.addTask(task2)
        taskManager.addTask(task3)

        self.assertEqual(3, len(taskManager.getAllTasks()))
        self.assertTrue(task1.getEndDate() > task2.getStartDate())
        self.assertTrue(task2.getEndDate() > task1.getStartDate())

    def testTaskCompletionUpdates(self):
        taskManager = TaskManager()
        task = Task(1, "Ibrahim", "Task for Completion", "Task to test completion updates.",
                    date(2024, 9, 1), date(2024, 9, 15))
        taskManager.addTask(task)

        task.markAsCompleted()
        task.setCompletionDate(date.today())

        self.assertTrue(task.isCompleted())
        self.assertIsNotNone(task.getCompletionDate())
        self.assertTrue(task.getCompletionDate() > task.getEndDate())

    def testSequentialTaskDependencies(self):
        taskManager = TaskManager()
        task1 = Task(1, "Ibrahim", "Task 1", "First task in a sequence.",
                    date(2024, 9, 1), date(2024, 9, 5))
        task2 = Task(2, "Kevin", "Task 2", "Dependent task.",
                    date(2024, 9, 6), date(2024, 9, 10))

        taskManager.addTask(task1)
        taskManager.addTask(task2)

        taskManager.markTaskAsCompleted(task1)

        self.assertEqual(1, len(taskManager.getCompletedTasks()))
        self.assertTrue(task2.getStartDate() > task1.getEndDate())

    def testTaskReassignment(self):
        taskManager = TaskManager()
        task = Task(1, "Ibrahim", "Reassign Task", "Task to test reassignment.",
                    date(2024, 9, 1), date(2024, 9, 15))
        taskManager.addTask(task)

        taskManager.changeOwner("Kevin", task)

        self.assertEqual("Kevin", task.getOwner())

    def testGetTasksOwnedBy(self):
        taskManager = TaskManager()
        task1 = Task(1, "Ibrahim", "Task 1", "Task owned by Ibrahim.",
                    date(2024, 9, 1), date(2024, 9, 15))
        task2 = Task(2, "Kevin", "Task 2", "Task owned by Kevin.",
                    date(2024, 9, 10), date(2024, 9, 20))
        task3 = Task(3, "Ibrahim", "Task 3", "Another task owned by Ibrahim.",
                    date(2024, 9, 5), date(2024, 9, 12))

        taskManager.addTask(task1)
        taskManager.addTask(task2)
        taskManager.addTask(task3)

        ibrahimTasks = taskManager.getTasksOwnedBy("Ibrahim")
        kevinTasks = taskManager.getTasksOwnedBy("Kevin")

        self.assertEqual(2, len(ibrahimTasks))
        self.assertEqual(1, len(kevinTasks))
        self.assertIn(task1, ibrahimTasks)
        self.assertIn(task3, ibrahimTasks)
        self.assertIn(task2, kevinTasks)

    def testUpdateOwnerAndGetTasksOwnedBy(self):
        taskManager = TaskManager()
        task1 = Task(1, "Ibrahim", "Task 1", "Task owned by Ibrahim.",
                    date(2024, 9, 1), date(2024, 9, 15))
        task2 = Task(2, "Kevin", "Task 2", "Task owned by Kevin.",
                    date(2024, 9, 10), date(2024, 9, 20))
        task3 = Task(3, "Ibrahim", "Task 3", "Another task owned by Ibrahim.",
                    date(2024, 9, 5), date(2024, 9, 12))

        taskManager.addTask(task1)
        taskManager.addTask(task2)
        taskManager.addTask(task3)

        ibrahimTasks = taskManager.getTasksOwnedBy("Ibrahim")
        kevinTasks = taskManager.getTasksOwnedBy("Kevin")

        self.assertEqual(2, len(ibrahimTasks))
        self.assertEqual(1, len(kevinTasks))
        self.assertIn(task1, ibrahimTasks)
        self.assertIn(task3, ibrahimTasks)
        self.assertIn(task2, kevinTasks)

        # let's update the owners
        taskManager.changeOwner("Ibrahim", task2)

        ibrahimTasks = taskManager.getTasksOwnedBy("Ibrahim")

        self.assertEqual(3, len(ibrahimTasks))
        self.assertIn(task1, ibrahimTasks)
        self.assertIn(task2, ibrahimTasks)
        self.assertIn(task3, ibrahimTasks)
